universal-application-tool-0.0.1/app/models/Program.java

```diff
@@ -54,6 +54,10 @@
   @OneToMany(mappedBy = "program")
   private List<Application> applications;
 
+  @ManyToMany
```

Comment:
I think you want this side of it to be "mappedBy" in the `ManyToMany` rather than having both sides hit the `JoinTable` (as with Versions and Programs).  The choice of ownership side is, I think, arbitrary.

Reply:
You're right! Thanks for catching that!